### PR TITLE
feat(Carta Giovani Nazionale): [#176443044] Adds directories skeleton for CGN implementation

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR creates the directories skeleton for CGN implementation and adds the base structure for store, reducers and navigation.

### CODE REVIEW

File: ts/features/bonus/cgn/navigation/routes.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const CGN_ROUTES = {
+  MAIN: "CGN_ROUTES_MAIN",
+
+  ACTIVATION: {
+    LOAD_CHECK_ACTIVATION_STATUS: "BPD_LOAD_CHECK_ACTIVATION_STATUS",
```
Comment (Undermaken) [inline ts/features/bonus/cgn/navigation/routes.ts:5]:
```suggestion
    LOAD_CHECK_ACTIVATION_STATUS: "CGN_LOAD_CHECK_ACTIVATION_STATUS",
```

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { availableBonusTypesSelectorFromId } from "../../../bonusVacanze/store/reducers/availableBonusesTypes";
+import { ID_CGN_TYPE } from "../../../bonusVacanze/utils/bonus";
+import BonusInformationComponent from "../../../common/components/BonusInformationComponent";
+import { navigateBack } from "../../../../../store/actions/navigation";
+
+export type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+/**
+ * This Screen shows all the information about the bpd program, with the rules and t&c.
```
Comment (Undermaken) [inline ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:16]:
```suggestion
 * This Screen shows all the information about the cgn program, with the rules and t&c.
```

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ */
+const CgnInformationScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const onConfirm = () => props.userActivateCgn();
+
+  return (
+    <>
+      {props.bonus ? (
+        <BonusInformationComponent
+          primaryCtaText={I18n.t("bonus.bpd.cta.activeBonus")}
```
Comment (Undermaken) [inline ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:25]:
```suggestion
          primaryCtaText={I18n.t("bonus.cgn.cta.activeBonus")}
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx (hunk 3 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:25]: I have just added this component as a draft (test were failing cause the CGN navigator had no routes inside) I will create relative ctas/texts and minor fixes on it's specific story https://www.pivotaltracker.com/story/show/176443163

File: ts/features/bonus/cgn/navigation/actions.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { NavigationActions } from "react-navigation";
+import CGN_ROUTES from "./routes";
+
+export const navigateToBpdOnboardingLoadActivationStatus = () =>
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/navigation/actions.ts:4]:
```suggestion
export const navigateToCgnOnboardingLoadActivationStatus = () =>
```

File: ts/features/bonus/cgn/navigation/actions.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  NavigationActions.navigate({
+    routeName: CGN_ROUTES.ACTIVATION.LOAD_CHECK_ACTIVATION_STATUS
+  });
+
+export const navigateToBpdOnboardingInformationTos = () =>
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/navigation/actions.ts:9]:
```suggestion
export const navigateToCgnOnboardingInformationTos = () =>
```

File: ts/features/bonus/cgn/navigation/actions.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  NavigationActions.navigate({
+    routeName: CGN_ROUTES.ACTIVATION.INFORMATION_TOS
+  });
+
+export const navigateToBpdOnboardingLoadActivate = () =>
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/navigation/actions.ts:14]:
```suggestion
export const navigateToCgnOnboardingLoadActivate = () =>
```

File: ts/features/bonus/cgn/navigation/actions.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  NavigationActions.navigate({
+    routeName: CGN_ROUTES.ACTIVATION.LOAD_ACTIVATE
+  });
+
+// Details
+export const navigateToBpdDetails = () =>
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/navigation/actions.ts:20]:
```suggestion
export const navigateToCgnDetails = () =>
```

**Review by fabriziofff — CHANGES_REQUESTED** [review]

**Review by fabriziofff — APPROVED** [review]

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx (hunk 3 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx]: @Undermaken I've yet included this fix in this PR with bc0e2f2

If you think everything's ok you can merge this PR

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+          onCancel={props.onCancel}
+          contextualHelp={emptyContextualHelp}
+        />
+      ) : null}
+    </>
+  );
+};
+
+const mapStateToProps = (state: GlobalState) => ({
+  // display the error with the retry only in case of networking errors
```
Comment (Undermaken) [inline ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:37]: is this comment related to the line below?

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+const CgnInformationScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const onConfirm = () => props.userActivateCgn();
+
+  return (
+    <>
+      {props.bonus ? (
+        <BonusInformationComponent
+          primaryCtaText={I18n.t("bonus.cgn.cta.activeBonus")}
+          bonus={props.bonus}
+          onConfirm={onConfirm}
```
Comment (Undermaken) [inline ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:27]: I don't know if onConfirm will be used few times, but at the moment you can void to create a new function
```suggestion
          onConfirm={props.userActivateCgn}
```

File: ts/features/bonus/cgn/store/reducers/activation.ts [hunk 10]
```diff
@@ -0,0 +1,47 @@
+// bonus reducer
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../../store/actions/types";
+import { cgnActivationStatus } from "../actions/activation";
+
+export enum CgnActivationProgressEnum {
+  "UNDEFINED" = "UNDEFINED",
```
Comment (Undermaken) [inline ts/features/bonus/cgn/store/reducers/activation.ts:7]: Since the reducer use only these states: `SUCCESS`, `PROGRESS`, `UNDEFINED`, `ERROR`
the other states could be removed? (if we'll need them, we can add them later)

or

You can replace these state with [RemoteValue](https://github.com/pagopa/io-app/blob/214f241851674d762a6c63e1a9a7ff9db06ab965/ts/features/bonus/bpd/model/RemoteValue.ts)?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx (hunk 9 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:27]: Still don't know at this time if we may include a check if the CGN is yet active and we'll need to show an alert to the user such we actually do on other bonuses

File: ts/features/bonus/cgn/store/reducers/activation.ts (hunk 10 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/store/reducers/activation.ts:7]: We're going to use all these values cause we need to request the activation status with a polling as we did with Bonus Vacanze, so even the PENDING/EXISTS values may be triggered.

The reducer will be evaluated properly when will be linked with the API call

Comment (CrisTofani) [reply ts/features/bonus/cgn/store/reducers/activation.ts:7]: eg: PENDING/EXISTS and other values will be dispatched by the related saga

File: ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx (hunk 8 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/screens/onboarding/CgnInformationScreen.tsx:37]: It should but i guess this it's not correct, if we have a network error we will notice on the bonuses list screen and not in the information one, i'm removing this comment

File: locales/en/index.yml [hunk 11]
```diff
@@ -2087,6 +2087,9 @@ bonus:
             active: "**The ranking changes every day.** The final one will be displayed on {{endDate}}."
             closed: "Ranking for this period is closed."
       lastUpdate: Updated at {{hour}} on {{date}}
+  cgn:
+    cta:
+      activeBonus: Attiva la tua Carta Giovani
```
Comment (Undermaken) [inline locales/en/index.yml:2092]:
```suggestion
      activeBonus: Activate your Carta Giovani
```

**Review by Undermaken — APPROVED** [review]
